crates/shim/src/synchronous/mod.rs

```diff
@@ -224,6 +281,22 @@ where
     }
 }
 
+#[cfg(windows)]
+fn create_server(flags: args::Flags) -> Result<Server> {
+    let mut server = Server::new();
+    let address = socket_address(&flags.address, &flags.namespace, &flags.id);
+    server = server.bind(address.as_str())?;
+    Ok(server)
+}
+
+#[cfg(unix)]
+fn create_server(_flags: args::Flags) -> Result<Server> {
+    let mut server = Server::new();
+    server = server.add_listener(SOCKET_FD)?;
```

Comment:
I'd prefer to have one function that works on both platforms, it's a bit easier to maintain (same applies for others)

```rust
fn create_server(_flags: args::Flags) -> Result<Server> {
     let mut server = Server::new();

     #[cfg(unix)]
     {
         server = server.add_listener(SOCKET_FD)?;
     }

     #[cfg(windows)]
     {
         ...
     }

     Ok(server)
 }
```

Reply:
I've converted most of these to a single function in https://github.com/containerd/rust-extensions/pull/139/commits/59f358fa1a799d786efc857f5f012d41dd65bd25

There were two places where I am not sure to handle it since the functions return different values:

```
#[cfg(windows)]
fn start_listener(address: &str) -> std::io::Result<()> {]

#[cfg(unix)]
fn start_listener(address: &str) -> std::io::Result<UnixListener> {
```

and the signal handling process stuff. 


Any pointers would be appreciated.

Reply:
I added a little helper struct with an `Option` wrapped around it to get around the limitation for the signal handling function. (70934a20fff1eb6570b845f1c03fa8bf92caa601). I think it was useful for the signal functions.  There is probably an opportunity to improve signal_hook to support windows for that API longer term. I could do it for the listener too but I am not sure it helps with code readability in that case.  

Let me know what you think.